### ISSUE TITLE
Simplify API

### ISSUE THREAD

**Issue by yukiisbored** [loklak/loklak_api_go#12]
The API is overly complicated and does not support the spirit of the Go programming language. Which results in high code complexity, confusing functions, etc.

**Comment by yukiisbored** [comment]
Also, using POST requests will be better than constructing complicated GET requests to decrease exploitation of the API

**Comment by aviaryan** [comment]
Agreed. For starters, we can create custom interfaces for each of the different return types (Apps, User, Tweets). Let's create individual issues and work on this step by step.

**Comment by yukiisbored** [comment]
@aviaryan I think it's better to make it into a simple API caller instead. Basically a communicator to keep it simple and flexible.

**Comment by aviaryan** [comment]
@yukiisbored Cool. Actually I was inspired from https://github.com/loklak/loklak_python_api which provides a much more coherent experience (like Twitter SDK). So if I want to do something like that, we will have to start a new project, right? The idea seems interesting to me and I will try it in my free time.

**Comment by yukiisbored** [comment]
@aviaryan Well, that's Python not Go. Go has a different taste, I suggest reading [Effective Go](https://golang.org/doc/effective_go.html) to see what I'm talking about.
